### PR TITLE
Make reviewer assignment use main branch version of vacation data

### PR DESCRIPTION
**Pull request by trodge**
Will fix https://github.com/hashicorp/terraform-provider-google/issues/19032 (currently testing)

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by trodge** [comment]
Closing to try re-opening

File: .ci/magician/github/membership.go [hunk 1]
```diff
@@ -177,6 +91,10 @@ func IsCoreContributor(user string) bool {
 }
 
 func IsCoreReviewer(user string) bool {
+	// TESTING PURPOSES
```
Comment (trodge) [inline .ci/magician/github/membership.go:94]: Not sure why this didn't work to make the bot auto-assign a reviewer for this PR.

**Review by c2thorn — APPROVED** [review]

File: .github/workflows/request-reviewer.yml [hunk 2]
```diff
@@ -24,6 +24,14 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.2
+      - name: Checkout membership data from main branch
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.2
+        with:
+          ref: main
+          path: .
```
Comment (trodge) [inline .github/workflows/request-reviewer.yml:31]: I *think* this should pull the new file from main and check it out over the old copy from whatever branch the PR was created with, but I'm not sure if `main` will always be GoogleCloudPlatform/main or potentially the fork's main.

File: .ci/magician/github/membership_data.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		{
+			id:        "roaks3",
+			startDate: newDate(2024, 8, 2, pdtLoc),
+			endDate:   newDate(2024, 8, 9, pdtLoc),
+		},
+		{
+			id:        "slevenick",
+			startDate: newDate(2024, 8, 10, pdtLoc),
+			endDate:   newDate(2024, 8, 17, pdtLoc),
+		},
```
Comment (trodge) [inline .ci/magician/github/membership_data.go:87]:
```suggestion
		},
				{
			id:        "trodge",
			startDate: newDate(2024, 8, 24, pdtLoc),
			endDate:   newDate(2024, 9, 2, pdtLoc),
		},
```

**Review by c2thorn — COMMENTED** [review]
Is this ready for review? Should I be the reviewer on it?

**Comment by trodge** [comment]
> Is this ready for review? Should I be the reviewer on it?

It should be ready, but I'm still not sure how to test if it actually works as intended. See https://github.com/GoogleCloudPlatform/magic-modules/pull/11445#discussion_r1718962621

Added Shuya in case we're still being cautious around CI changes.

File: .github/workflows/request-reviewer.yml (hunk 2 above)
Comment (shuyama1) [reply .github/workflows/request-reviewer.yml:31]: Can we confirm this refers to GoogleCloudPlatform/main? Otherwise, I'd assume it will fail for most of the PRs.

**Review by shuyama1 — COMMENTED** [review]
Mostly LGTM. Is there a way to test the changes before merging?

File: .github/workflows/request-reviewer.yml (hunk 2 above)
Comment (trodge) [reply .github/workflows/request-reviewer.yml:31]: It will have the same `repository` as the previous step, which should be whichever repo the PR is opened on. [See Doc](https://github.com/actions/checkout?tab=readme-ov-file#usage)

**Comment by trodge** [comment]
> Mostly LGTM. Is there a way to test the changes before merging?

I'm testing it in my fork and finding that it breaks the workflow. Going to try a few more variations of the sparse checkout.

File: .github/workflows/request-reviewer.yml [hunk 4]
```diff
@@ -24,6 +24,14 @@ jobs:
...
+          sparse-checkout: |
+            .ci/magician/github/membership_data.go
+          sparse-checkout-cone-mode: false
```
Comment (trodge) [inline .github/workflows/request-reviewer.yml:34]:
```suggestion
        with:
          ref: main
```

Based on discussion, I think we just want to checkout everything from main.

See https://github.com/trodge/magic-modules/pull/14
https://github.com/trodge/magic-modules/actions/runs/10725729173/job/29744244495

Feel free to open a PR with trodge/magic-modules as base if you can think of any edge cases I didn't test.

**Review by shuyama1 — APPROVED** [review]
Thanks for testing it!
